src/chromium/interfaces/launchBrowser.js

```diff
@@ -4,21 +4,26 @@ import { isCurrentTabValidUrlScheme } from "Shared/backgroundScripts/validTab.js
 /**
  * Launches the Firefox variant. If Firefox is not installed, launch the Firefox download page.
  *
- * @param {*} tab The tab object to launch the browser with.
+ * @param {string} url The url to launch in the browser.
  * @param {boolean} usePrivateBrowsing  True if firefox should be launched in private browsing mode, false for normal mode.
  * @returns {boolean} True if the browser was launched, false otherwise.
  */
-export async function launchBrowser(tab, usePrivateBrowsing = false) {
+export async function launchBrowser(url, usePrivateBrowsing = false) {
   if (!(await getIsFirefoxInstalled())) {
     browser.tabs.create({ url: "https://www.mozilla.org/firefox/" });
     return false;
   }
 
   if (isCurrentTabValidUrlScheme) {
```

Comment:
The updated logic (which looks up whatever the current tab is) highlights the flaw of the `isCurrentTabValidUrlScheme` flag that is part of the validTab module, which I mentioned before at https://github.com/mozilla/firefox-launch/pull/18/files#r1468673433

There is no guaranteed relation between the validation of validTab.js and the URL opened here. I recommend to validate the URL here as a last stop before handing it off to the protocol handler.

Bonus tip: use the targetUrlPatterns on the link menu, and documentUrlPatterns on the page menu (i.e. the `contextMenus.create` calls for them) to make sure that the menu item only appears on context menus targeting supported schemes.

Reply:
I've added a check within launchBrowser, but will get to replacing the old logic entirely in a future patch